### PR TITLE
Basic test for CWL

### PR DESCRIPTION
**Pull request by philtweir**
This brings in a basic test for a toy CWL workflow.

### CODE REVIEW

File: src/dewret/workflow.py [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+
+    name: str
+    target: Lazy
+
+    def __init__(self, name: str, target: Lazy):
+        """Initialize the Task.
+
+        Argument:
+            name: Name of wrapped function.
+            callable: Actual function being wrapped (optional).
```
Comment (elleryames) [inline src/dewret/workflow.py:84]: Looks like this has been renamed to `target`.

File: src/dewret/__main__.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""CLI for dewret.
+
+Simple CLI for running a workflow. It is more likely to use this
```
Comment (elleryames) [inline src/dewret/__main__.py:17]: > Simple CLI for running a workflow. ...

My understanding is that the main purpose of this tooling, and also this CLI, is to render a declarative workflow in a particular format such as CWL. Is the description "running a workflow" misleading?

File: src/dewret/workflow.py [hunk 3]
```diff
@@ -0,0 +1,281 @@
...
+
+        Starts from a result, and builds a workflow to output it.
+        """
+        step = result.step
+        workflow = result.__workflow__
+        workflow.set_output(result)
+        return workflow
+
+    def set_output(self, output: StepReference) -> None:
+        """Choose the output step.
```
[the remark this answers is not in the document]
Comment (philtweir) [reply src/dewret/workflow.py:173]: No, this is simpler - it is the result you request that triggers the graph to run, so equivalent to the thing you call compute on in dask, for example. Updated the name to `result` as that's more in line with dask terminology (and less conflicting with our common usage too).

File: src/dewret/workflow.py (hunk 1 above)
Comment (philtweir) [reply src/dewret/workflow.py:84]: Correct - updated.

**Comment by philtweir** [comment]
AP: add ruff docstring checking

**Pull request merged**